zigzag/classes/hardware/architecture/imc_unit.py

```diff
@@ -0,0 +1,196 @@
+import math
+if __name__ == "__main__":
+    from get_cacti_cost import get_cacti_cost
+else:
+    from zigzag.classes.hardware.architecture.get_cacti_cost import get_cacti_cost
+
+###############################################################################################################
+# This file includes:
+#   . class LogicUnit (defines the energy/area/delay cost of multipliers, adders, regs)
+#   . class ImcArray (provides initialization function, used for class DimcArray and AimcArray)
+###############################################################################################################
+
+class LogicUnit:
```

Comment:
If you define the technology data separately in a _technology_ folder, you can here then just pass the technology number and directly use that.

Reply:
The technology data will be passed from the user input.